src/ewma_by_time.rs

```diff
@@ -18,13 +18,27 @@ pub(crate) fn impl_ewma_by_time_float(
         TimeUnit::Nanoseconds => half_life * 1_000,
     };
 
-    let mut prev_time: i64 = times.get(0).unwrap();
-    let mut prev_result = values.get(0).unwrap();
-    out.push(Some(prev_result));
+    let mut skip_rows: usize = 0;
+    let mut prev_time: i64 = 0;
+    let mut prev_result: f64 = 0.;
+    for (idx, value) in values.iter().enumerate() {
+        match value {
+            Some(value) => {
+                prev_time = times.get(idx).unwrap();
```

Comment:
isn't this going to panic if `times[idx]` is `None`? do we need to zip over both `values` and `times` here?

Reply:
Yes, it will; I've fixed.